Tests/TestingTests/ClockTests.swift

```diff
@@ -119,4 +120,15 @@ struct ClockTests {
 #endif
     #expect(duration == .nanoseconds(offsetNanoseconds))
   }
+
+  @available(_clockAPI, *)
+  @Test("Codable")
+  func codable() async throws {
+    let instant = Test.Clock.Instant().advanced(by: .nanoseconds(100))
+    let decoded = try JSONDecoder().decode(Test.Clock.Instant.self,
+                                           from: JSONEncoder().encode(instant))
+
+    #expect(instant == decoded)
+    #expect(instant != Test.Clock.Instant())
```

Comment:
Probably unlikely, but could this potentially fail and become flaky if exactly `.nanoseconds(100)` elapses during this test? Maybe this second `!=` expectation should be done in a way that first captures the "now" value in a `let`

Reply:
We could instead encode and decode fixed-value instants, yeah?